feature-account-impl/src/main/java/io/novafoundation/nova/feature_account_impl/data/proxy/RealProxySyncService.kt

```diff
@@ -0,0 +1,166 @@
+
+import io.novafoundation.nova.common.address.AccountIdKey
+import io.novafoundation.nova.common.address.intoKey
+import io.novafoundation.nova.common.utils.CollectionDiffer
+import io.novafoundation.nova.core_db.dao.MetaAccountDao
+import io.novafoundation.nova.core_db.model.chain.account.ChainAccountLocal
+import io.novafoundation.nova.core_db.model.chain.account.MetaAccountLocal
+import io.novafoundation.nova.core_db.model.chain.account.ProxyAccountLocal
+import io.novafoundation.nova.feature_account_api.data.model.ProxiedWithProxies
+import io.novafoundation.nova.feature_account_api.data.proxy.ProxySyncService
+import io.novafoundation.nova.feature_account_api.data.repository.ProxyRepository
+import io.novafoundation.nova.feature_account_api.domain.account.identity.Identity
+import io.novafoundation.nova.feature_account_api.domain.account.identity.IdentityProvider
+import io.novafoundation.nova.feature_account_api.domain.interfaces.AccountRepository
+import io.novafoundation.nova.feature_account_api.domain.model.LightMetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccount
+import io.novafoundation.nova.feature_account_api.domain.model.MetaAccountId
+import io.novafoundation.nova.feature_account_api.domain.model.accountIdIn
+import io.novafoundation.nova.runtime.ext.addressOf
+import io.novafoundation.nova.runtime.ext.isSubstrateBased
+import io.novafoundation.nova.runtime.multiNetwork.ChainRegistry
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
+import io.novafoundation.nova.runtime.multiNetwork.findChains
+import jp.co.soramitsu.fearless_utils.runtime.AccountId
+
+class RealProxySyncService(
+    private val chainRegistry: ChainRegistry,
+    private val proxyRepository: ProxyRepository,
+    private val accounRepository: AccountRepository,
+    private val accountDao: MetaAccountDao,
+    private val identityProvider: IdentityProvider
+) : ProxySyncService {
+
+    override suspend fun startSyncing() {
+        if (!accounRepository.hasMetaAccounts()) return
+
+        val metaAccounts = getMetaAccounts()
+
+        val supportedProxyChains = getSupportedProxyChains()
+        val chainsToAccountIds = supportedProxyChains.associateWith { chain -> chain.getAvailableAccountIds(metaAccounts) }
+
+        // proxiedsWithProxies will be usefull when we union differen proxy types to one account
+        val proxiedsWithProxies = chainsToAccountIds.flatMap { (chain, accountIds) ->
+            proxyRepository.getProxyDelegatorsForAccounts(chain.id, accountIds)
+        }
+
+        val newProxies = proxiedsWithProxies.formatToLocalProxies()
+        val oldProxies = accountDao.getAllProxyAccounts()
+
+        val proxiesDiff = CollectionDiffer.findDiff(newProxies, oldProxies, forceUseNewItems = false)
+
+        insertMetaAndChainAccounts(proxiesDiff)
+        insertProxies(proxiesDiff)
```

Comment:
This can be atomic by combining both calls into one `@Transaction` annotated method

Reply:
I changed this logic in another PR, since I need to have proxied meta id in ProxyAccountLocal and it seems it's not possible to use Collection differ here